### PR TITLE
New: Emit annotation thread events to the viewer and beyond

### PR DESCRIPTION
**Pull request by pramodsum**
Emits annotator and annotation thread specific events to the Viewer and Preview

Annotator Events: 
- annotationmodeenter: Enter an annotation mode
- annotationmodeexit: Exit an annotation mode
- annotationsfetched: Annotations have been fetched from the API
- annotationerror: Annotation error occurred

Annotation Thread Events:
- annotationpending: Thread was created but not saved
- annotationthreadsaved: Annotation thread saved on the server
- annotationthreaddeleted: Annotation thread deleted on the server
- annotationsaved: Annotation saved on an existing thread on the server
- annotationdeleted: Annotation deleted from an existing thread on the server
- annotationdeleteerror: Error while deleting an annotation on either an existing/new thread
- annotationcanceled: Annotation cancelled from posting on either an existing/new thread
- annotationcreateerror: Error while creating an annotation on either an existing/new thread

### CODE REVIEW

**Comment by pramodsum** [comment]
@varunmaker the events listed above will be emitted from Annotations

File: src/lib/annotations/AnnotationThread.js [hunk 1]
```diff
@@ -149,30 +155,24 @@ class AnnotationThread extends EventEmitter {
...
      * @param {boolean} [useServer] - Whether or not to delete on server, default true
-     * @return {void}
+     * @return {Promise} - Annotation delete promise
      */
     deleteAnnotation(annotationID, useServer = true) {
         // Ignore if no corresponding annotation exists in thread or user doesn't have permissions
         const annotation = this.annotations.find((annot) => annot.annotationID === annotationID);
         if (!annotation || (annotation.permissions && !annotation.permissions.can_delete)) {
-            return;
+            return Promise.resolve();
```
Comment (JustinHoldstock) [inline src/lib/annotations/AnnotationThread.js]: Could this be a Promise.reject() with a message that the annotation either "does not exist" or "cannot be deleted"?

Comment (pramodsum) [reply src/lib/annotations/AnnotationThread.js]: Oh. Yea i think that's totally fair. I'm not sure it make sense to reject the promise later on in this method if `useServer = false` but here i think it makes sense. You don't have permission to do what you're trying to do so therefore should get an error

File: src/lib/annotations/Annotator.js [hunk 2]
```diff
@@ -669,7 +668,7 @@ class Annotator extends EventEmitter {
         if (!service || !(service instanceof AnnotationService)) {
             return;
         }
-        service.removeAllListeners('annotatorerror');
+        service.removeAllListeners(ANNOTATOR_EVENT.error);
```
Comment (jeremypress) [inline src/lib/annotations/Annotator.js]: `removeListeners` + the function name

File: src/lib/annotations/Annotator.js [hunk 3]
```diff
@@ -705,10 +694,7 @@ class Annotator extends EventEmitter {
      * @return {void}
      */
     unbindCustomListenersOnThread(thread) {
-        thread.removeAllListeners('threaddeleted');
-        thread.removeAllListeners('threadcleanup');
-        thread.removeAllListeners('annotationsaved');
-        thread.removeAllListeners('annotationevent');
+        thread.removeAllListeners('threadevent');
```
Comment (jeremypress) [inline src/lib/annotations/Annotator.js]: I think if you remove the args this should work. Make sure they're actually getting removed though

**Pull request merged**